### ISSUE TITLE
Infinite loop because of followlinks=True

### ISSUE THREAD

**Issue by Minipada** [TimesysGit/vigiles-buildroot#8]
I am working on integrating vigiles in buildroot but I have one package (acl and host-acl) which are on an infinite loop.

This package, provided by buildroot has symlinks linking to the parent folder. Why, I don't know, but it's there.

```bash
ls -lisah vendor/buildroot/output-legacy/build/acl-2.3.2/include/acl
70217887 4.0K lrwxrwxrwx 1 david.bensoussan domain users 91 Aug 29 12:51 vendor/buildroot/output-legacy/build/acl-2.3.2/include/acl -> /home/david.bensoussan/dev/m-vigiles/vendor/buildroot/output-legacy/build/acl-2.3.2/include

ls -lisah vendor/buildroot/output-legacy/build/acl-2.3.2/include/sys
70217888 4.0K lrwxrwxrwx 1 david.bensoussan domain users 91 Aug 29 12:51 vendor/buildroot/output-legacy/build/acl-2.3.2/include/sys -> /home/david.bensoussan/dev/m-vigiles/vendor/buildroot/output-legacy/build/acl-2.3.2/include
```

Then, when iterating through the folder, it infinitely iterates, see this line:

https://github.com/TimesysGit/vigiles-buildroot/blob/3e281e651d2ce02d25e7e522407f75663206e40b/scripts/packages.py#L203

Given it's a default buildroot recipe we didn't change, I am not sure how to approach the problem, any suggestion? Do we need the followlinks=True ?

**Comment by harish-bansal** [comment]
Thanks @Minipada for reporting this issue.  We are able to reproduce this issue.

`os.walk` is skipped for `output` directory , but your directory name is `output-legacy` that causes an infinite loop. 
We are working on a fix for it. In the interim I suggest you use `output` as directory name.

**Comment by Minipada** [comment]
Hi, good to know. Thanks for the fast answer!
We have many targets which start with output- . I can't really change that as we're a mid size company and it would disturb the workflow of too many.
Is there a way to dynamically get the output directory name and skip it? Thank you!

**Comment by harish-bansal** [comment]
> Hi, good to know. Thanks for the fast answer! We have many targets which start with output- . I can't really change that as we're a mid size company and it would disturb the workflow of too many. Is there a way to dynamically get the output directory name and skip it? Thank you!

We are working on it and I'll share the patch with you once ready.

**Comment by harish-bansal** [comment]
Hi @Minipada,

Please try the attached patch file for skipping the output directory. Hope it unblocks you for now !
[exclude_output_dir.patch.txt](https://github.com/user-attachments/files/16813508/exclude_output_dir.patch.txt)

**Comment by Minipada** [comment]
This is working perfectly! Can this be merged into master?

**Comment by harish-bansal** [comment]
> This is working perfectly! Can this be merged into master?

Glad to hear that! Our test team needs to test and approve this fix. Post their regression we'll push it to master.

**Comment by Minipada** [comment]
Excellent, thanks for the support

**Comment by harish-bansal** [comment]
The fix https://github.com/TimesysGit/vigiles-buildroot/commit/f4976520a5066ce654433e2bb40617fa02e2ede4 has been pushed to the master branch
